### ISSUE TITLE
Specified file does not contain the proper metadata blocks when in fact it does

### ISSUE THREAD

**Issue by pke** [OpenUserJS/OpenUserJS.org#2024]
### What is missing?

I have run into "Specified file does not contain the proper metadata blocks." while importing a user script that actually contains all the metadata blocks that exist.
However it was not suffixed `.user.js` as greaseyfork does not have such requirement when importing scripts.

### Why?

The 400 error message is misleading. If openuserjs requires all user scripts to end in `.user.js` then it should not _offer_ to import misnamed files (exclude them from the view or make them shallow rendered) from github **or** change the error message to give the user a hint.

**Comment by Martii** [comment]
Thanks for the report.

Please reread https://openuserjs.org/about/Userscript-Beginners-HOWTO#what-is-a-user-script- vs. https://openuserjs.org/about/Userscript-Beginners-HOWTO#what-is-a-library-script-

>  then it should not offer to import misnamed files

... and knock out the way we do importing of actual libraries I think not.

Thanks though.

**Comment by pke** [comment]
Thank for the quick response. So the error message could still be improved in stating that the importer thinks to handle a library script and have it list the metadata blocks its missing?

**Comment by Martii** [comment]
> importer thinks to handle a library script

That's one use case but the inverse is also true. If a .user.js is contains a library script *(which is never should be and is again user error)* then message is clear as it can be. **Plus the author clicked Add Script or Add Library** so if the user has read the basics on how to do .user.js then they know to name it the appropriate way. If we ever support any additional types then that would be a tertiary test. We are still considering how to do #1971 , which is similar, but future translations are a factor there. Embedding "extra" info that the Author should have already read provides a challenge to translations in the current ecosystem.

Long story short if an Author has say 200 .user.js and we were to load in all of them just to mark what type they were not only would GitHub rate limit EVERYONE as a result they could actually block us from importing. 

So it's best if the Author reads the basic HOWTO document first and does it on their own.
